extensions/machine-learning-services/src/common/utils.ts

```diff
@@ -0,0 +1,54 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as uuid from 'uuid';
+import * as path from 'path';
+import * as os from 'os';
+import * as fs from 'fs-extra';
+import * as constants from '../common/constants';
+
+export async function execCommandOnTempFile<T>(content: string, command: (filePath: string) => Promise<T>): Promise<T> {
+	let tempFilePath: string;
+	try {
+		tempFilePath = path.join(os.tmpdir(), `ads_ml_temp_${uuid.v4()}`);
+		await fs.writeFile(tempFilePath, content);
+		let result = await command(tempFilePath);
+		return result;
+	}
+	catch (err) {
+		throw err;
+	}
+	finally {
+		fs.unlink(tempFilePath);
+	}
+}
+
+export async function exists(path: string): Promise<boolean> {
```

Comment:
why not fs.existsSync?

Reply:
FYI it's preferred not to use the sync methods when we can and instead use the promise versions of these methods in the fs module (this lets the thread do other stuff while it's waiting for disk access)

Reply:
because I want to do sync

---
In reply to: [356364273](https://github.com/microsoft/azuredatastudio/pull/8622#discussion_r356364273) [](ancestors = 356364273)

Reply:
Then you'd await on the Promise version. The reason sync is bad because it blocks the thread the entire time it's doing disk I/O. With await the execution of that code will stop until it finishes but it allows the thread to do other things in the background while it waits.

Reply:
I changed it already to use fs.promise. is it something else I'm missing? 

---
In reply to: [357307052](https://github.com/microsoft/azuredatastudio/pull/8622#discussion_r357307052) [](ancestors = 357307052)

Reply:
Nope, I just hadn't looked at the changes yet!

Reply:
You you don't like the try catch weirdness of access or lsstat you can just promisify the not sync version of exists.